pkg/backend/metrics.go

```diff
@@ -0,0 +1,203 @@
+			dimensions["customisksizegb"] = strconv.FormatInt(int64(p.DiskSizeGB), 10)
+		}
+
+		if p.EncryptionAtHost == api.EncryptionAtHostEnabled {
+			dimensions["workprofiles.encryptionathost"] = fmt.Sprintf("%s.%s", string(api.EncryptionAtHostEnabled), p.Name)
+		} else if p.EncryptionAtHost == api.EncryptionAtHostDisabled {
+			dimensions["workprofiles.encryptionathost"] = fmt.Sprintf("%s.%s", string(api.EncryptionAtHostDisabled), p.Name)
+		} else {
+			dimensions["workprofiles.encryptionathost"] = unknown
+		}
+	}
+
+	if doc.OpenShiftCluster.Properties.ClusterProfile.FipsValidatedModules == api.FipsValidatedModulesEnabled {
+		dimensions["fips"] = string(api.FipsValidatedModulesEnabled)
+	} else if doc.OpenShiftCluster.Properties.ClusterProfile.FipsValidatedModules == api.FipsValidatedModulesDisabled {
+		dimensions["fips"] = string(api.FipsValidatedModulesDisabled)
+	} else {
+		dimensions["fips"] = unknown
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherAuthMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	if doc.OpenShiftCluster.Properties.PlatformWorkloadIdentityProfile != nil {
+		dimensions["workloadidentity"] = enabled
+	} else if doc.OpenShiftCluster.Properties.ServicePrincipalProfile != nil {
+		dimensions["serviceprincipal"] = disabled
+	} else {
+		dimensions["serviceprincipal"] = unknown
+	}
+
+	if doc.OpenShiftCluster.Properties.ClusterProfile.PullSecret != "" {
+		dimensions["pullsecret"] = enabled
+	} else {
+		dimensions["pullsecret"] = disabled
+	}
+}
+
+func (ocb *openShiftClusterBackend) gatherNetworkMetrics(doc *api.OpenShiftClusterDocument, dimensions map[string]string) {
+	for _, p := range doc.OpenShiftCluster.Properties.IngressProfiles {
+		if p.Visibility == api.VisibilityPrivate {
+			dimensions["ingressprofile"] = fmt.Sprintf("%s.%s", string(api.VisibilityPrivate), p.Name)
+		} else if p.Visibility == api.VisibilityPublic {
+			dimensions["ingressprofile"] = fmt.Sprintf("%s.%s", string(api.VisibilityPublic), p.Name)
+		} else {
+			dimensions["ingressprofile"] = unknown
+		}
+	}
+
+	if doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType == api.OutboundTypeUserDefinedRouting {
+		dimensions["networkprofile.outboundtype"] = string(api.OutboundTypeUserDefinedRouting)
+	} else if doc.OpenShiftCluster.Properties.NetworkProfile.OutboundType == api.OutboundTypeLoadbalancer {
+		dimensions["networkprofile.outboundtype"] = string(api.OutboundTypeLoadbalancer)
+	} else {
+		dimensions["networkprofile.outboundtype"] = unknown
+	}
+
+	if doc.OpenShiftCluster.Properties.NetworkProfile.PodCIDR != "10.128.0.0/14" {
+		dimensions["podcidr"] = custom
+	} else {
+		dimensions["podcidr"] = defaultSet
+	}
+
+	if doc.OpenShiftCluster.Properties.NetworkProfile.ServiceCIDR != "172.30.0.0/16" {
+		dimensions["servicecidr"] = custom
+	} else {
+		dimensions["servicecidr"] = defaultSet
+	}
+
+	if doc.OpenShiftCluster.Properties.ClusterProfile.Domain != "" {
+		dimensions["domain"] = enabled
+	}
```

Comment:
`domain` is always set to a value during cluster installations, either a fully qualified domain provided by the customer, or a random ID for managed domains. 

We may want to include logic similar to https://github.com/Azure/ARO-RP/blob/6997415b87d29fa466963f7c50a71c02122045da/pkg/util/dns/dns.go#L195 to identify whether the domain is managed or not.

Reply:
Thanks, I updated this to check if it's a managed domain or not.